### PR TITLE
fix(dbauth-mw): Unset cookie instead of clearing

### PR DESCRIPTION
**Pull request by dac09**
Updates dbAuth middleware implementation to _unset_ the cookies, instead of clearing them.

This fixes the case where the cookie should be removed when decryption fails.

### CODE REVIEW

File: packages/auth-providers/dbAuth/middleware/src/__tests__/createDbAuthMiddleware.test.ts [hunk 1]
```diff
@@ -522,6 +522,7 @@ describe('createDbAuthMiddleware()', () => {
 
       const serverAuthContext = req.serverAuthContext.get()
       expect(serverAuthContext).toBeNull()
+      expect(res.cookies.entries()).toBe('bazinga')
```
Comment (dac09) [inline packages/auth-providers/dbAuth/middleware/src/__tests__/createDbAuthMiddleware.test.ts:525]: Haven't finished this test obviously... @dthyresson if you want to feel free to finish the test up while I sleep.

If not I'll tag you again tomorrow.

**Comment by dac09** [comment]
@dthyresson will leave for you to merge just so you're aware. We were missing part of the test case here, otherwise would've caught it!

File: packages/auth-providers/dbAuth/middleware/src/__tests__/createDbAuthMiddleware.test.ts [hunk 2]
```diff
@@ -522,6 +522,14 @@ describe('createDbAuthMiddleware()', () => {
 
       const serverAuthContext = req.serverAuthContext.get()
       expect(serverAuthContext).toBeNull()
+
+      // mwResponse gets converted to a response eventually.
+      const finalResponse = res.toResponse()
+      expect(finalResponse.headers.getSetCookie()).toEqual([
```
Comment (Tobbe) [inline packages/auth-providers/dbAuth/middleware/src/__tests__/createDbAuthMiddleware.test.ts:528]:
```suggestion
      // Verify that the cookies were unset
      expect(finalResponse.headers.getSetCookie()).toEqual([
```

File: packages/auth-providers/dbAuth/middleware/src/__tests__/createDbAuthMiddleware.test.ts [hunk 3]
```diff
@@ -522,6 +522,14 @@ describe('createDbAuthMiddleware()', () => {
...
+        // Expired cookies here!
```
Comment (Tobbe) [inline packages/auth-providers/dbAuth/middleware/src/__tests__/createDbAuthMiddleware.test.ts:529]:
```suggestion
        // Expired cookies with no values
```

File: packages/auth-providers/dbAuth/middleware/src/__tests__/createDbAuthMiddleware.test.ts (hunk 2 above)
Comment (Tobbe) [inline packages/auth-providers/dbAuth/middleware/src/__tests__/createDbAuthMiddleware.test.ts:528]: To be honest the code comment you had was confusing me more than it was helping me. I suggest just getting rid of it and inlining the `res.toResponse()` part. I think the code speaks for itself here 🙂 

```suggestion
      expect(res.toResponse().headers.getSetCookie()).toEqual([
```

**Review by dthyresson — COMMENTED** [review]
Leaving as is in favor of @Tobbe 's comments

**Review by Tobbe — APPROVED** [review]

**Pull request merged**